code/datums/components/material_container.dm

```diff
@@ -53,7 +53,7 @@
 	var/list/tc = allowed_typecache
 	if(user.a_intent != INTENT_HELP)
 		return
-	if((I.flags_2 & (HOLOGRAM_2 | NO_MAT_REDEMPTION_2)) || (tc && !is_type_in_typecache(I, tc)))
+	if((I.flags_2 & HOLOGRAM_2 || I.item_flags & NO_MAT_REDEMPTION) || (tc && !is_type_in_typecache(I, tc)))
```

Comment:
Precedence rules are such that this should be fine, but parens would help readers to not have to check